src/components/Charts/PriceChart/index.tsx

```diff
@@ -0,0 +1,312 @@
+
+  const { ticks, tickTimestampFormatter, crosshairTimestampFormatter } = useMemo(() => {
+    // Limits the number of ticks based on graph width
+    const maxTicks = Math.floor(dimensions.width / 100)
+
+    const ticks = getTicks(chart.startingPrice.timestamp, chart.endingPrice.timestamp, timePeriod, maxTicks)
+    const tickTimestampFormatter = getTimestampFormatter(timePeriod, locale, TimestampFormatterType.TICK)
+    const crosshairTimestampFormatter = getTimestampFormatter(timePeriod, locale, TimestampFormatterType.CROSSHAIR)
+
+    return { ticks, tickTimestampFormatter, crosshairTimestampFormatter }
+  }, [dimensions.width, chart.startingPrice.timestamp, chart.endingPrice.timestamp, timePeriod, locale])
+
+  const theme = useTheme()
+  const [crosshair, setCrosshair] = useState<{ x: number; y: number; price: PricePoint }>()
+  const resetCrosshair = useCallback(() => setCrosshair(undefined), [setCrosshair])
+
+  const setCrosshairOnHover = useCallback(
+    (event: Element | EventType) => {
+      const { x } = localPoint(event) || { x: 0 }
+      const price = getNearestPricePoint(x, prices, timeScale)
+
+      if (price) {
+        const x = timeScale(price.timestamp)
+        const y = priceScale(price.value)
+        setCrosshair({ x, y, price })
+      }
+    },
+    [priceScale, timeScale, prices]
+  )
+
+  // Resets the crosshair when the time period is changed, to avoid stale UI
+  useEffect(() => resetCrosshair(), [resetCrosshair, timePeriod])
+
+  const crosshairEdgeMax = dimensions.width * 0.85
+  const crosshairAtEdge = !!crosshair && crosshair.x > crosshairEdgeMax
+
+  // Default curve doesn't look good for the HOUR chart.
+  // Higher values make the curve more rigid, lower values smooth the curve but make it less "sticky" to real data points,
+  // making it unacceptable for shorter durations / smaller variances.
+  const curveTension = timePeriod === TimePeriod.HOUR ? 1 : 0.9
+
+  const getX = useCallback((p: PricePoint) => timeScale(p.timestamp), [timeScale])
+  const getY = useCallback((p: PricePoint) => priceScale(p.value), [priceScale])
+  const curve = useMemo(() => curveCardinal.tension(curveTension), [curveTension])
+
+  return (
+    <>
+      <ChartHeader chart={chart} crosshairPrice={crosshair?.price} />
+      <svg data-cy="price-chart" width={dimensions.width} height={dimensions.height} style={{ minWidth: '100%' }}>
+        <AnimatedInLineChart
+          data={prices}
+          getX={getX}
+          getY={getY}
+          marginTop={dimensions.marginTop}
+          curve={curve}
+          strokeWidth={2}
+        />
+        {blanks.map((blank, index) => (
+          <FadedInLineChart
+            key={index}
+            data={blank}
+            getX={getX}
+            getY={getY}
+            marginTop={dimensions.marginTop}
+            curve={curve}
+            strokeWidth={2}
+            color={theme.neutral3}
+            dashed
+          />
+        ))}
+        {crosshair !== undefined ? (
```

Comment:
this used to be `crosshair !== null` checking this is intentional

Reply:
yes, when I first wrote this code I chose null for some reason, but now I'm an undefined-maxi